### PR TITLE
feat:[CDS-80838]: add api for fetching all infra accessible at a scope

### PR DESCRIPTION
**Pull request by sourabh-awashti**
## Describe your changes
Changes are regarding the addition of new api for fetching the infras accessible at any scope.
tests Done:
1. Fetched all infrastructure at project/org/account level.
2. List Api for fetching Infra for a particular env.
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55047)

### CODE REVIEW

**Comment by sourabh-awashti** [comment]
trigger uts

**Review by lovish1999 — DISMISSED** [review]
LGTM

**Comment by sourabh-awashti** [comment]
trigger uts

File: 120-ng-manager/src/test/java/io/harness/ng/core/infrastructure/resource/InfrastructureResourceTest.java [hunk 1]
```diff
@@ -132,6 +158,41 @@ public void testUpsertServiceWithSchemaValidationFlagOn() throws IOException {
...
+                                     .yaml("yaml")
+                                     .build();
+    Page<InfrastructureEntity> infraEntities = new PageImpl<>(Arrays.asList(infra), pageable, 1);
+    when(cdFeatureFlagHelper.isEnabled(ACCOUNT_ID, FeatureName.DISABLE_CDS_SERVICE_ENV_SCHEMA_VALIDATION))
+        .thenReturn(false);
+
+    when(cdFeatureFlagHelper.isEnabled(ACCOUNT_ID, FeatureName.CDS_SCOPE_INFRA_TO_SERVICES)).thenReturn(false);
+    when(infrastructureEntityService.list(any(), any())).thenReturn(infraEntities);
+    doNothing().when(accessControlClient).checkForAccessOrThrow(any(), any(), any());
+    doReturn(AccessCheckResponseDTO.builder()
```
Comment (tchaurasiya) [inline 120-ng-manager/src/test/java/io/harness/ng/core/infrastructure/resource/InfrastructureResourceTest.java:180]: nit: Lets add multiple infra in test set (with different env at different scope, and some with permitted false to ensure our filtering logic)

**Review by tchaurasiya — CHANGES_REQUESTED** [review]

File: 120-ng-manager/src/main/java/io/harness/ng/core/infrastructure/resource/InfrastructureResource.java [hunk 2]
```diff
@@ -617,4 +624,105 @@ private void validateProjectLevelInfraScope(InfrastructureEntity entity) {
...
+  @Path("/list-access")
+  @ApiOperation(value = "Gets Infrastructure access list ", nickname = "getInfrastructureAccessList")
+  @Operation(operationId = "getInfrastructureAccessList", summary = "Gets Infrastructure access list",
+      responses =
+      {
+        @io.swagger.v3.oas.annotations.responses.
+        ApiResponse(description = "Returns the list of Infrastructure accessible at the current scope")
+      })
+  @Hidden
+  public ResponseDTO<List<InfrastructureResponse>>
```
[the remark this answers is not in the document]
Comment (sourabh-awashti) [reply 120-ng-manager/src/main/java/io/harness/ng/core/infrastructure/resource/InfrastructureResource.java:613]: done

File: 120-ng-manager/src/main/java/io/harness/ng/core/infrastructure/resource/InfrastructureResource.java [hunk 3]
```diff
@@ -617,4 +624,105 @@ private void validateProjectLevelInfraScope(InfrastructureEntity entity) {
...
+  listAccessInfrastructures(@Parameter(description = NGCommonEntityConstants.PAGE_PARAM_MESSAGE) @QueryParam(
```
[the remark this answers is not in the document]
Comment (sourabh-awashti) [reply 120-ng-manager/src/main/java/io/harness/ng/core/infrastructure/resource/InfrastructureResource.java:614]: done

File: 120-ng-manager/src/test/java/io/harness/ng/core/infrastructure/resource/InfrastructureResourceTest.java (hunk 1 above)
Comment (sourabh-awashti) [reply 120-ng-manager/src/test/java/io/harness/ng/core/infrastructure/resource/InfrastructureResourceTest.java:183]: done

**Comment by sourabh-awashti** [comment]
trigger uts

**Review by tchaurasiya — DISMISSED** [review]

**Review by lovish1999 — APPROVED** [review]
LGTM

**Comment by sourabh-awashti** [comment]
trigger uts

**Comment by sourabh-awashti** [comment]
trigger codeformat

**Review by lovish1999 — APPROVED** [review]

**Review by tchaurasiya — APPROVED** [review]

**Comment by sourabh-awashti** [comment]
trigger review

**Comment by sourabh-awashti** [comment]
trigger messagecheck

**Comment by sourabh-awashti** [comment]
trigger messagecheck

**Comment by sourabh-awashti** [comment]
trigger messagecheck

**Pull request merged**